### PR TITLE
New Wazuh module to collect Github audit logs

### PR DESCRIPTION
**Pull request by TomasTurina**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/7660|

## Description

GitHub module implementation to collect API logs.

WIP.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
  - [x] Windows
  - [x] MAC OS X
- [x] Source installation
- [x] Package installation
- [x] Source upgrade
- [x] Package upgrade
- [x] Review logs syntax and correct language

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [x] Valgrind

- [x] Configuration on demand reports new parameters
- [x] Added unit tests (for new features)

### CODE REVIEW

File: ruleset/rules/0750-github_rules.xml [hunk 1]
```diff
@@ -0,0 +1,3225 @@
...
+    <rule id="91120" level="3">
+        <if_sid>91100</if_sid>
+        <action type="osregex">^dependabot_security_updates.\.*</action>
+        <description>GitHub Dependabot security updates.</description>
+        <options>no_full_log</options>
+        <group>git_dependabot_security_updates</group>
+    </rule>
+
+    <!-- dependabot_security_updates.disable -->
+    <rule id="91121" level="9">
```
Comment (fabamatic) [inline ruleset/rules/0750-github_rules.xml:225]: level 9 is "Error from invalid source", perhaps level 12 "High importance event" is more appropiate

File: ruleset/rules/0750-github_rules.xml [hunk 2]
```diff
@@ -0,0 +1,3225 @@
...
+        <if_sid>91120</if_sid>
+        <action>dependabot_security_updates.disable</action>
+        <description>GitHub Dependabot security updates disable.</description>
+        <options>no_full_log</options>
+        <group>git_dependabot_security_updates</group>
+    </rule>
+
+    <!-- dependabot_security_updates.enable -->
+    <rule id="91122" level="5">
```
Comment (fabamatic) [inline ruleset/rules/0750-github_rules.xml:234]: level 5 is "User generated error", perhaps level 3 "Successful/Authorized events" is more appropiate

File: ruleset/rules/0750-github_rules.xml [hunk 3]
```diff
@@ -0,0 +1,3225 @@
...
+    <rule id="91123" level="3">
+        <if_sid>91100</if_sid>
+        <action type="osregex">^dependabot_security_updates_new_repos.\.*</action>
+        <description>GitHub Dependabot security updates new repos.</description>
+        <options>no_full_log</options>
+        <group>git_dependabot_security_updates_new_repos</group>
+    </rule>
+
+    <!-- dependabot_security_updates_new_repos.disable -->
+    <rule id="91124" level="9">
```
Comment (fabamatic) [inline ruleset/rules/0750-github_rules.xml:255]: level 9 is "Error from invalid source", perhaps level 12 "High importance event" is more appropriate

File: ruleset/rules/0750-github_rules.xml [hunk 4]
```diff
@@ -0,0 +1,3225 @@
...
+        <if_sid>91123</if_sid>
+        <action>dependabot_security_updates_new_repos.disable</action>
+        <description>GitHub Dependabot security updates new repos disable.</description>
+        <options>no_full_log</options>
+        <group>git_dependabot_security_updates_new_repos</group>
+    </rule>
+
+    <!-- dependabot_security_updates_new_repos.enable -->
+    <rule id="91125" level="5">
```
Comment (fabamatic) [inline ruleset/rules/0750-github_rules.xml:264]: level 5 is "User generated error", perhaps level 3 "Successful/Authorized events" is more appropriate

File: ruleset/rules/0750-github_rules.xml [hunk 5]
```diff
@@ -0,0 +1,3225 @@
...
+    <rule id="91126" level="3">
+        <if_sid>91100</if_sid>
+        <action type="osregex">^dependency_graph.\.*</action>
+        <description>GitHub Dependency graph.</description>
+        <options>no_full_log</options>
+        <group>git_dependency_graph</group>
+    </rule>
+
+    <!-- dependency_graph.disable -->
+    <rule id="91127" level="9">
```
Comment (fabamatic) [inline ruleset/rules/0750-github_rules.xml:285]: level 9 is "Error from invalid source", perhaps level 12 "High importance event" is more appropriate

File: ruleset/rules/0750-github_rules.xml [hunk 6]
```diff
@@ -0,0 +1,3225 @@
...
+    <rule id="91139" level="3">
+        <if_sid>91100</if_sid>
+        <action type="osregex">^enterprise.\.*</action>
+        <description>GitHub Enterprise.</description>
+        <options>no_full_log</options>
+        <group>git_enterprise</group>
+    </rule>
+
+    <!-- enterprise.remove_self_hosted_runner -->
+    <rule id="91140" level="7">
```
Comment (fabamatic) [inline ruleset/rules/0750-github_rules.xml:412]: level 7 should be used for “Bad word” matching, not sure what does removing self hosted runner implies

File: ruleset/rules/0750-github_rules.xml [hunk 7]
```diff
@@ -0,0 +1,3225 @@
...
+        <description>GitHub Environment Remove protection rule.</description>
+        <options>no_full_log</options>
+        <group>git_environment</group>
+    </rule>
+
+    <!-- git
+            clone
+            fetch
+            push
+    -->
```
Comment (fabamatic) [inline ruleset/rules/0750-github_rules.xml:576]: Does this mean that an alert will be generated for each clone, fetch, and push? Perhaps level should be lower that 3 for these rules?

Comment (fabamatic) [reply ruleset/rules/0750-github_rules.xml:576]: Ok, reading #7660 its clear that these events need to be stored as alerts in order to monitor and generate dashboards and so on. Perhaps we should be thinking about a new level used only for log audits, with no semantic value, in other words we are not trying to detect an attack, we are just storing audit logs from Github API

File: ruleset/rules/0750-github_rules.xml [hunk 8]
```diff
@@ -0,0 +1,3225 @@
...
+    <rule id="91272" level="3">
+        <if_sid>91100</if_sid>
+        <action type="osregex">^private_repository_forking.\.*</action>
+        <description>GitHub Private repository forking.</description>
+        <options>no_full_log</options>
+        <group>git_private_repository_forking</group>
+    </rule>
+
+    <!-- private_repository_forking.enable -->
+    <rule id="91273" level="5">
```
Comment (fabamatic) [inline ruleset/rules/0750-github_rules.xml:1718]: Looks like enabling forking of a private repository should have higher level than disabling forking on a private repository

**Review by fabamatic — COMMENTED** [review]
Only checked and verified files 0750-github_rules.xml and github.ini.
If we consider these to be detection rules there may be some considerations to be made about rule levels for example why would a billing_plan_change event generate a level 9 alert? The most probable thing is that the action would have been a legit one.
If we consider these rules to be useful just for audit purposes then adding a semantic value in the rules is not needed or at least they may be ignored, having a level 5, 7 or 3 would be more or less the same if the rules are just a means to count and measure the events.  
Anyways, I think that the most accurate level for all these rules would be 3 "Successful/Authorized events" because as far as we know the audit API reports on those kind of events, they are not error or security events.

Wazuh documentation on rules classification:https://documentation.wazuh.com/current/user-manual/ruleset/rules-classification.html

Edit: Reading the API events on more detail, there are some that may be more interesting from a security perspective

- codespaces
- dependabot_alerts
- dependabot_security_updates
- org_credential_authorization
- oauth_application
- repository_vulnerability_alert
- repository_vulnerability_alerts

File: ruleset/rules/0750-github_rules.xml (hunk 2 above)
Comment (cborla) [reply ruleset/rules/0750-github_rules.xml]: Done

File: ruleset/rules/0750-github_rules.xml (hunk 3 above)
Comment (cborla) [reply ruleset/rules/0750-github_rules.xml]: Done

File: ruleset/rules/0750-github_rules.xml (hunk 4 above)
Comment (cborla) [reply ruleset/rules/0750-github_rules.xml]: Done

File: ruleset/rules/0750-github_rules.xml (hunk 5 above)
Comment (cborla) [reply ruleset/rules/0750-github_rules.xml]: Done

File: ruleset/rules/0750-github_rules.xml (hunk 8 above)
Comment (cborla) [reply ruleset/rules/0750-github_rules.xml]: Done

File: ruleset/rules/0750-github_rules.xml (hunk 7 above)
Comment (cborla) [reply ruleset/rules/0750-github_rules.xml:576]: Assigned level 3 for clone, fetch and push events.

**Review by fabamatic — DISMISSED** [review]

**Review by 72nomada — APPROVED** [review]
LGTM

**Review by 72nomada — DISMISSED** [review]
LGTM

**Review by vikman90 — APPROVED** [review]

**Pull request merged**